test/e2e/common.go

```diff
@@ -83,20 +88,24 @@ func dumpSpecResourcesAndCleanup(ctx context.Context, specName string, clusterPr
 	})
 
 	if !skipCleanup {
```

Comment:
wdyt about:

```go
	defer func() {
		cancelWatches()
		redactLogs()
	}

	if skipCleanup {
		return
	}

	if cluster == nil {
		log.Logf("Unable to delete workload cluster as cluster is nil")
	}
	
	Byf("Deleting cluster %s/%s", cluster.Namespace, cluster.Name)
	// While https://github.com/kubernetes-sigs/cluster-api/issues/2955 is addressed in future iterations, there is a chance
	// that cluster variable is not set even if the cluster exists, so we are calling DeleteAllClustersAndWait
	// instead of DeleteClusterAndWait
	framework.DeleteAllClustersAndWait(ctx, framework.DeleteAllClustersAndWaitInput{
		Client:    clusterProxy.GetClient(),
		Namespace: namespace.Name,
	}, intervalsGetter(specName, "wait-delete-cluster")...)

	Byf("Deleting namespace used for hosting the %q test spec", specName)
	framework.DeleteNamespace(ctx, framework.DeleteNamespaceInput{
		Deleter: clusterProxy.GetClient(),
		Name:    namespace.Name,
	})
```

Reply:
Nice, I think that reads more cleanly with `defer`. I'll refactor it.